core/src/main/java/io/substrait/relation/Cross.java

```diff
@@ -1,10 +1,21 @@
 package io.substrait.relation;
 
+import io.substrait.type.Type;
+import io.substrait.type.TypeCreator;
+import java.util.stream.Stream;
 import org.immutables.value.Value;
 
 @Value.Immutable
 public abstract class Cross extends BiRel {
 
+  @Override
+  protected Type.Struct deriveRecordType() {
```

Comment:
I've realised that adding this removes the `deriveRecordType` method from the `ImmutableCross.Builder`, meaning this is technically a breaking change.